frame/ethereum/src/lib.rs

```diff
@@ -473,6 +483,17 @@ impl<T: Config> Pallet<T> {
 		let transaction_data: TransactionData = transaction.into();
 		let transaction_nonce = transaction_data.nonce;
 
+		let (weight_limit, proof_size_base_cost) =
+			match <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+				transaction_data.gas_limit.unique_saturated_into(),
+				true,
+			) {
+				weight_limit if weight_limit.proof_size() > 0 => {
+					(Some(weight_limit), Some(Self::transaction_len(transaction)))
+				}
+				_ => (None, None),
+			};
```

Comment:
This code block is used twice in this file. Can we reuse it and hide in behind the `evm-with-weight-limit` feature?

for example:
```rs
if `evm-with-weight-limit` enable {
      let (weight_limit, proof_size_base_cost) = xxx;
} else {
  (None, None)
}
```

Reply:
And the `transaction_len` only used under this feature.

Reply:
No feature anymore, resolving